js/services/contact_service.js

```diff
@@ -251,6 +251,10 @@ angular.module('contactsApp')
 		if (contact.addressBookId === addressbook.displayName) {
 			return;
 		}
+		if(addressbook.readOnly) {
+			OC.Notification.showTemporary(t('contacts', 'You don\'t have permission to write to this addressbook.'));
```

Comment:
It should be 'You don't have permission to write to this addressbook.'

Reply:
If you're talking about the `\'` it's to escape the ' character since we're using it to define the string. :)